src/test/java/org/magicdgs/thaplv/tools/GetHaplotypesIntegrationTest.java

```diff
@@ -0,0 +1,104 @@
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package org.magicdgs.thaplv.tools;
+
+
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.test.IntegrationTestSpec;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public class GetHaplotypesIntegrationTest extends CommandLineProgramTest {
+
+    @Test
+    public void testGetHaplotypesHaploid() throws IOException {
+        IntegrationTestSpec testSpec = new IntegrationTestSpec(
+                " -V " + getTestDataDir() + "example.vcf"
+                        + " -H HAPLOID"
+                        + " -O %s",
+                Arrays.asList(getTestDataDir() + "example_haploid.vcf")
+        );
+        testSpec.executeTest("testGetHaplotypesHaploid", this);
+    }
+
+    @Test
+    public void testGetHaplotypesOutputPloidy1() throws IOException {
+        IntegrationTestSpec testSpec = new IntegrationTestSpec(
+                " -V " + getTestDataDir() + "example.vcf"
+                        + " -H HAPLOID"
+                        + " -ploidy 1"
+                        + " -O %s",
+                Arrays.asList(getTestDataDir() + "example_haploid_1.vcf")
+        );
+        testSpec.executeTest("testGetHaplotypesOutputPloidy1", this);
+    }
+
+    @Test
+    public void testGetHaplotypesBackCross() throws IOException {
+        IntegrationTestSpec testSpec = new IntegrationTestSpec(
+                " -V " + getTestDataDir() + "example.vcf"
+                        + " -H BACK_CROSS"
+                        + " -O %s",
+                Arrays.asList(getTestDataDir()
+                        + "example_backcross.vcf")
+        );
+        testSpec.executeTest("testGetHaplotypesBackCross", this);
+    }
+
+    @Test
+    public void testGetHaplotypesDontCheckError() throws IOException {
+        IntegrationTestSpec testSpec = new IntegrationTestSpec(
+                " -H DONT_CHECK"
+                        + " -V " + getTestDataDir() + "example.vcf"
+                        + " -O %s",
+                1,
+                UserException.class
+        );
+        testSpec.executeTest("testGetHaplotypesDontCheckError", this);
+    }
+
+    @Test
+    public void testGetHaplotypesWrongPloidy() throws IOException {
+        IntegrationTestSpec testSpec = new IntegrationTestSpec(
+                " -H DONT_CHECK"
```

Comment:
This should be a different model, to be sure that it is the ploidy the one that throw the UserException.